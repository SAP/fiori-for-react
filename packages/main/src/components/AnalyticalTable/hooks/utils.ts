export const makeTemplateColumns = (columns, resizedColumns) =>
  columns
    .map((column) => {
      if (resizedColumns[column.id]) {
        return `${resizedColumns[column.id]}px`;
      }
      return `minmax(${column.minWidth}px, ${column.width || '1fr'})`;
    })
    .join(' ');
