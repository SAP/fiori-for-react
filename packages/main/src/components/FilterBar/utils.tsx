import { cloneElement, ReactElement } from 'react';

export const setPropsOfChildren = (children, currentRef) => {
  return children.map((child) => {
    if (child.props.children.props[currentRef]?.children) {
      const childrenOfChild = child.props.children;
      if (
        child.props.children.props[currentRef]?.tagName === 'UI5-SELECT' ||
        child.props.children.props[currentRef]?.tagName === 'UI5-MULTI-COMBOBOX'
      ) {
        const selectedIndices = Array.from(child.props.children.props[currentRef].children)
          .map((item, index) => (item._state.selected ? index : false))
          .filter((el) => el !== false);
        const options = child.props.children.props.children.map((item, index) => {
          if (selectedIndices.includes(index)) {
            return cloneElement(item, { selected: true });
          }
          return cloneElement(item, { selected: false });
        });
        return cloneElement(child, {
          children: { ...childrenOfChild, props: { ...childrenOfChild.props, children: options } }
        });
      }
      if (
        child.props.children.props[currentRef]?.tagName === 'UI5-INPUT' ||
        child.props.children.props[currentRef]?.tagName === 'UI5-DATEPICKER'
      ) {
        return cloneElement(child, {
          children: {
            ...childrenOfChild,
            props: { ...childrenOfChild.props, value: child.props.children.props[currentRef]._state.value }
          }
        });
      }
      if (
        child.props.children.props[currentRef]?.tagName === 'UI5-SWITCH' ||
        child.props.children.props[currentRef]?.tagName === 'UI5-CHECKBOX'
      ) {
        return cloneElement(child, {
          children: {
            ...childrenOfChild,
            props: { ...childrenOfChild.props, checked: child.props.children.props[currentRef]._state.checked }
          }
        });
      }
      return child;
    }
    return child;
  });
};

export const addRef = (children, refsArray, currentRef) =>
  children.map((child) => {
    const childRef = refsArray.filter((item) => item.key === child.key);
    const childrenAttributes = child.props.children;
    const childrenProps = child.props.children.props;
    return cloneElement(child as ReactElement<any>, {
      children: { ...childrenAttributes, props: { ...childrenProps, [currentRef]: childRef[0]?.node } }
    });
  });

export const renderSearchWithValue = (searchElement, searchValue) => {
  return cloneElement(searchElement(), { value: searchValue });
};
