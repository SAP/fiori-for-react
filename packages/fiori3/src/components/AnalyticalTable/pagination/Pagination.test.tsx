import React from 'react';
import sinon from 'sinon';
import { Pagination, LinkHOC } from './index';
import { mountThemedComponent } from '@shared/tests/utils';

const defaultProps = {
  numItems: 100,
  pages: 10,
  page: 1,
  canNext: true,
  canPrevious: true
};

describe('Table/Pagination', () => {
  test('With props', () => {
    // @ts-ignore
    const wrapper = mountThemedComponent(<Pagination {...defaultProps} />);
    expect(wrapper.render()).toMatchSnapshot();
  });

  test('Showing middle numbers', () => {
    // @ts-ignore
    const wrapper = mountThemedComponent(<Pagination {...defaultProps} page={5} />);
    expect(wrapper.render()).toMatchSnapshot();
  });

  test('Click handling', () => {
    const onSelectionChange = sinon.spy();
    const dummyOnPageChange = () => null; // method is generally being provided by react-table
    const wrapper = mountThemedComponent(
      // @ts-ignore
      <Pagination {...defaultProps} page={5} onSelectionChange={onSelectionChange} onPageChange={dummyOnPageChange} />
    );
    const links = wrapper.find(LinkHOC);
    links.at(0).simulate('click');
    links.at(3).simulate('click');
    links.at(links.length - 1).simulate('click');
    links.at(1).simulate('click');
    expect(onSelectionChange.callCount).toBe(3);
  });
});
