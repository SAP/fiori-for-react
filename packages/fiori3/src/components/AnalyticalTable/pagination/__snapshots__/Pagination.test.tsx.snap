// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Table/Pagination Showing middle numbers 1`] = `
<div
  class="Pagination-container-"
>
  <div
    class="Pagination-total-"
  >
    <span
      class="Label-label- Label-label-"
    >
      100 Items
    </span>
  </div>
  <a
    class="Link-link-"
    href="javascript:void(0)"
    target="_blank"
  >
    &lt;
  </a>
  <a
    class="Link-link-"
    href="javascript:void(0)"
    target="_blank"
  >
    1
  </a>
  <a
    class="Link-link-"
    href="javascript:void(0)"
    target="_blank"
  >
    ...
  </a>
  <a
    class="Link-link-"
    href="javascript:void(0)"
    target="_blank"
  >
    5
  </a>
  <a
    class="Link-link- Link-emphasized-"
    href="javascript:void(0)"
    target="_blank"
  >
    6
  </a>
  <a
    class="Link-link-"
    href="javascript:void(0)"
    target="_blank"
  >
    7
  </a>
  <a
    class="Link-link-"
    href="javascript:void(0)"
    target="_blank"
  >
    ...
  </a>
  <a
    class="Link-link-"
    href="javascript:void(0)"
    target="_blank"
  >
    10
  </a>
  <a
    class="Link-link-"
    href="javascript:void(0)"
    target="_blank"
  >
    &gt;
  </a>
</div>
`;

exports[`Table/Pagination With props 1`] = `
<div
  class="Pagination-container-"
>
  <div
    class="Pagination-total-"
  >
    <span
      class="Label-label- Label-label-"
    >
      100 Items
    </span>
  </div>
  <a
    class="Link-link-"
    href="javascript:void(0)"
    target="_blank"
  >
    &lt;
  </a>
  <a
    class="Link-link-"
    href="javascript:void(0)"
    target="_blank"
  >
    1
  </a>
  <a
    class="Link-link- Link-emphasized-"
    href="javascript:void(0)"
    target="_blank"
  >
    2
  </a>
  <a
    class="Link-link-"
    href="javascript:void(0)"
    target="_blank"
  >
    3
  </a>
  <a
    class="Link-link-"
    href="javascript:void(0)"
    target="_blank"
  >
    ...
  </a>
  <a
    class="Link-link-"
    href="javascript:void(0)"
    target="_blank"
  >
    10
  </a>
  <a
    class="Link-link-"
    href="javascript:void(0)"
    target="_blank"
  >
    &gt;
  </a>
</div>
`;
